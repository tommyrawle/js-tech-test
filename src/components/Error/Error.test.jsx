import React from 'react';
import { shallow } from 'enzyme';

let wrapper;
const mockHistoryPush = jest.fn();
const mockProps = {
  setError: jest.fn(),
  error: { code: 500, message: 'something bad happened' }
};
describe('Error component 1st code block', () => {
  beforeEach(() => {
    jest.resetModules();
    jest.mock('react-router-dom', () => ({
      ...jest.requireActual('react-router-dom'),
      useHistory: () => ({
        push: mockHistoryPush
      }),
      useLocation: () => ({
        pathname: '/error'
      })
    }));
    const { Error } = require('./Error.jsx');
    wrapper = shallow(<Error {...mockProps} />);
  });
  it('should render 1st block in Error component', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('if the pathname is /error, it should render the first block', () => {
    const code = wrapper
      .find('h1')
      .at(1)
      .text();
    expect(parseInt(code)).toBe(500);
    expect(wrapper.find('p').text()).toBe('something bad happened');
  });
});
describe('Error component 2nd code block', () => {
  beforeEach(() => {
    jest.resetModules();
    jest.mock('react-router-dom', () => ({
      ...jest.requireActual('react-router-dom'),
      useHistory: () => ({
        push: mockHistoryPush
      }),
      useLocation: () => ({
        pathname: '/cheese'
      })
    }));
    const { Error } = require('./Error.jsx');
    wrapper = shallow(<Error {...mockProps} />);
  });
  it('should render 2nd block in Error component', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('if the pathname is NOT /error, it should render the second block', () => {
    const code = wrapper
      .find('h1')
      .at(1)
      .text();
    expect(code).toBe('404');
    expect(wrapper.find('p').text()).toBe('Page Not Found');
  });
});

describe('back button', () => {
  beforeEach(() => {
    jest.resetModules();
    jest.mock('react-router-dom', () => ({
      ...jest.requireActual('react-router-dom'),
      useHistory: () => ({
        push: mockHistoryPush
      }),
      useLocation: () => ({
        pathname: '/cheese'
      })
    }));
    const { Error } = require('./Error.jsx');
    wrapper = shallow(<Error {...mockProps} />);
  });
  it('should call mockHistoryPush on click', () => {
    wrapper.find('button').simulate('click');
    expect(mockHistoryPush).toBeCalled();
  });
});
