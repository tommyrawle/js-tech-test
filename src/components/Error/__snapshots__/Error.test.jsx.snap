// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error component 1st code block should render 1st block in Error component 1`] = `
<styled.div>
  <div>
    <div>
      <h1>
        Oops!
      </h1>
      <h3>
        It seems something has gone wrong
      </h3>
      <h1>
        500
      </h1>
      <p>
        something bad happened
      </p>
    </div>
    <button
      onClick={[Function]}
    >
      Head Back
    </button>
  </div>
</styled.div>
`;

exports[`Error component 2nd code block should render 2nd block in Error component 1`] = `
<styled.div>
  <div>
    <div>
      <h1>
        Oops!
      </h1>
      <h3>
        It seems you took a wrong turn.
      </h3>
      <h1>
        404
      </h1>
      <p>
        Page Not Found
      </p>
    </div>
    <button
      onClick={[Function]}
    >
      Head Back
    </button>
  </div>
</styled.div>
`;
